### PR TITLE
Disallow null attribute array values.

### PR DESCRIPTION
**Pull request by anuraaga**
Thanks to https://github.com/open-telemetry/opentelemetry-specification/commit/06638fc65519a021e29a213e54feae30880ce39e#diff-500cb43b9725373c25ea03660c293e59681e6e2f6ba3eb588188ed3b83dcbdcd

This is not ABI compatible but is API compatible for practical usages of the methods so I didn't do any deprecation.

### CODE REVIEW

**Review by thisthat — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]
Thanks! This is a good tightening up of the API.

**Comment by jkwatson** [comment]
weird. the build is failing on a completely unrelated deprecation warning.

File: api/src/main/java/io/opentelemetry/api/common/AttributesBuilder.java [hunk 1]
```diff
@@ -155,4 +155,28 @@ public AttributesBuilder putAll(Attributes attributes) {
     attributes.forEach(this::put);
     return this;
   }
+
+  private static List<Double> toList(double... values) {
```
Comment (carlosalberto) [inline api/src/main/java/io/opentelemetry/api/common/AttributesBuilder.java:159]: Oh well. I wish Java had **real** generics :/

**Review by carlosalberto — APPROVED** [review]

**Comment by jkwatson** [comment]
needs a rebase to pick up the merge-conflict fix

File: api/src/main/java/io/opentelemetry/api/common/AttributesBuilder.java (hunk 1 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/api/common/AttributesBuilder.java:159]: I'm sure we'll get them with java 21.  😂

**Pull request merged**